src/LocalAppController.ts

```diff
@@ -71,15 +71,14 @@ export class LocalAppController {
         targetConfig = await resolveDebugConfigurationWithSubstitutedVariables(targetConfig);
         app.jmxPort = parseJMXPort(targetConfig.vmArgs);
 
-        const cwdUri: vscode.Uri = vscode.Uri.parse(app.path);
         const launchConfig = Object.assign({}, targetConfig, {
-            noDebug: !debug,
-            cwd: cwdUri.fsPath,
+            noDebug: !debug
         });
         if (profile) {
             launchConfig.vmArgs = launchConfig.vmArgs + ` -Dspring.profiles.active=${profile}`;
         }
 
+        const cwdUri: vscode.Uri = vscode.Uri.parse(launchConfig.cwd);
```

Comment:
Since the variable `cwdUri` is just used to find the workspace root at the later place `vscode.workspace.getWorkspaceFolder(cwdUri)`, I think keeping `const cwdUri: vscode.Uri = vscode.Uri.parse(app.path);` is enough. Your change `const cwdUri: vscode.Uri = vscode.Uri.parse(launchConfig.cwd);` doesn't work if the user uses a variable `${workspaceFolder}` as `cwd` in launch.json.

So the question is how to support an explicit `cwd` when launching springboot app. 

Look at the issue https://github.com/microsoft/vscode-spring-boot-dashboard/issues/71, where it requires the submodule in its own folder. So we need to find a balance to satisfy these two needs.


```

Reply:
@testforstephen 
What exactly won't work with using a variable like `${workspaceFolder}` in launch.json? I'm using this variable and starting and debugging of application works fine. Is there an additional feature/integration I have to a take a look at?

Reply:
At this moment, the variable `${workspaceFolder}` in `launchConfig.cwd` is not populated yet. Using `vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(launchConfig.cwd))` is equivalent to `vscode.workspace.getWorkspaceFolder(vscode.Uri.parse("${workspaceFolder}"))`, and will always return a value `undefined`.

Reply:
Maybe I got it. So if we would put `app.path` as folder into the debug configuration, but don't overwrite it in the launch configuration, we should be able to serve both use cases, aren't we?
Then we should be able to use something like `${workspaceFolder}/..` to target the parent dir and in default it would be looked up to app path. 
I'll test that and push an update.